hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -87,6 +87,7 @@
    * set
    */
   private volatile boolean closed;
+  private S3Object object;
```

Comment:
can you add a javadoc comment explaining why this is needed "we have to keep a reference to this to stop garbage collection breaking the input stream"